src/table/utils/copy-utils.ts

```diff
@@ -1,9 +1,20 @@
 const copyCellValue = async (evt: any) => {
+  let value: string | undefined | null = '';
   const target = evt.target as HTMLElement;
 
-  const value = target.children[0]?.firstChild
-    ? target.children[0].firstChild.textContent
-    : target.firstChild?.textContent;
+  if (target.classList.contains('sn-table-head-cell')) {
+    value = target.children[0].firstChild?.textContent;
+  }
+  if (!target.classList.contains('sn-table-head-cell') && target.classList.contains('sn-table-cell')) {
+    value = target.firstChild?.textContent;
+  }
+  if (target.classList.contains('MuiTableSortLabel-root')) {
+    value = target.firstChild && target.firstChild.textContent;
+  }
+  if (target.classList.contains('MuiTableSortLabel-icon')) {
+    value = target.parentElement && target.parentElement.firstChild?.textContent;
+  }
+
```

Comment:
This solution might not be a nice one but the only way I could cover all possible click-area.